backend-k8s/steward-system/101-customResourceDefinition_PipelineRun.yaml

```diff
@@ -85,30 +85,24 @@ spec:
               pattern: '^run|abort$'
             logging: ###
               type: object
-              required:
-                - elasticsearch
               properties:
                 elasticsearch: ###
                   type: object
                   required:
                     - runID
                   properties:
                     runID: ###
-                      type: object
+                      type: object # should be any JSON value
             runDetails: ###
               type: object
-              required:
```

Comment:
Is this handled accordingly in the code? I was not sure.

Reply:
If it's not, then it's a bug. I'll check it.

Reply:
We check if runDetails is nil, but we take the parameters as they are. I assume `tektonStringParam` or `fmt.Sprintf` might fail with a nil parameter - need to check.
https://github.com/SAP/stewardci-core/blob/master/pkg/runctl/run_manager.go#L254
```go
	details := spec.RunDetails
	if details != nil {
		params := []tekton.Param{
			tektonStringParam("JOB_NAME", details.JobName),
			tektonStringParam("RUN_NUMBER", fmt.Sprintf("%d", details.SequenceNumber)),
			tektonStringParam("RUN_CAUSE", details.Cause),
		}
		tektonTaskRun.Spec.Inputs.Params = append(tektonTaskRun.Spec.Inputs.Params, params...)
	}
```

Reply:
Anyway, I think it would be better to *not* set the StringParam if it is nil.
And we should check if the JenkinsfileRunner (and other components?) handle not set parameters correctly.

Reply:
See #44.

The JFR image entrypoint script can handle the absence or emptyness of each of the run detail parameters.